Expecto/Expecto.fs

```diff
@@ -810,7 +810,42 @@ module Impl =
           do! innerPrinter.summary c s
           tcLog "testSuiteFinished" [
             "name", "ExpectoTestSuite" ] } }
+    static member fromInterface (i : ITestPrinter) =
+      { beforeRun   = i.BeforeRun >> Async.AwaitTask
+        beforeEach  = i.BeforeEach >> Async.AwaitTask
+        passed      = fun n d -> i.Passed(n, d) |> Async.AwaitTask
```

Comment:
This makes the async run-once instead of run-many. Better to do `async { return i.BeforeRun }` or the like (delaying the construction of the `Task` value).